File: SideMenuDemo/SideMenuDemo/ViewControllers/SideMenuViewController/SideMenuViewController.m
//
//  SideMenuViewController.m
//  SideMenu
//
//  Created by Vincent on 2016/12/27.
//  Copyright © 2016年 Vincent. All rights reserved.
//

#import "SideMenuViewController.h"
#import "TestViewController.h"
#import "UIViewController+SideMenu.h"
@interface SideMenuViewController ()<UITableViewDelegate,UITableViewDataSource>

@property (nonatomic, weak) IBOutlet UITableView *tableView;

@property (nonatomic, strong) NSArray *sourceArray;
@end

@implementation SideMenuViewController


#pragma mark lazy loading 
- (NSArray *)sourceArray{
    
    if (_sourceArray == nil) {
        _sourceArray = @[@"menu",@"menu",@"menu",@"menu",@"menu",@"menu",@"menu",@"menu",@"menu",@"menu",@"menu",@"menu",@"menu",@"menu",@"menu",@"menu",@"menu",@"menu",@"menu"];
    }
    
    return _sourceArray;
}
- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view from its nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

#pragma mark tableViewDelegate

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section{
    
    return self.sourceArray.count;
}
- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath{
    
    
    UITableViewCell*cell = [tableView dequeueReusableCellWithIdentifier:@"SideMenuViewController"];
    
    if (cell == nil) {
        cell = [[UITableViewCell alloc]initWithStyle:UITableViewCellStyleDefault reuseIdentifier:@"SideMenuViewController"];
    }
    
    
    cell.textLabel.text = self.sourceArray [indexPath.row];
    return cell;
}
- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath{
    
    return 44;
}
- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath{
    
    [tableView deselectRowAtIndexPath:indexPath animated:YES];
    [self sideMenuSlideIn];
    
    
    [(UINavigationController*) [UIApplication sharedApplication].keyWindow.rootViewController  pushViewController:[TestViewController new] animated:YES];
    
    
}
@end

File: VTSideMenuSwiftDemo/VTSideMenuSwiftDemo/SideMenuViewController.swift
//
//  SideMenuViewController.swift
//  VTSideMenuSwiftDemo
//
//  Created by Vincent on 2017/1/4.
//  Copyright © 2017年 Vincent. All rights reserved.
//

import UIKit

class SideMenuViewController: UIViewController,UITableViewDataSource,UITableViewDelegate {

    
    @IBOutlet weak var tableView: UITableView!
    
    private let sourceArray = ["menu","menu","menu","menu","menu","menu","menu","menu","menu","menu","menu","menu","menu","menu","menu","menu"]

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    // MARK: tableviewDelegate,tableviewDatasource
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        
        return sourceArray.count;
    }
    
    
    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
        
        return 44;
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        
        var cell =  tableView.dequeueReusableCell(withIdentifier: "SideMenuViewController")
        
        if cell == nil {
            
            cell = UITableViewCell(style: UITableViewCellStyle.default, reuseIdentifier: "SideMenuViewController")
            
        }
        
        cell?.textLabel?.text = sourceArray[indexPath.row]
        
        return cell!;
        
        
    }
    
    
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        
        tableView.deselectRow(at: indexPath, animated: true)
        (UIApplication.shared.keyWindow?.rootViewController as!UINavigationController) .pushViewController(TestViewController(), animated: true)
        self.sideMenuSlideIn()
        
        
        
        
        
    }
    
    
    
    //cell seperator will start from x:0 step 2
    func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
        
        if cell.responds(to: #selector(setter: cell.layoutMargins)) {
            
            cell.layoutMargins = UIEdgeInsets.zero
            
        }
        
        if cell.responds(to: #selector(setter: cell.separatorInset)) {
            cell.separatorInset = UIEdgeInsets.zero
        }
        
        
        
    }
    
    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */

}
